corgi/tasks/pnc.py

```diff
@@ -123,3 +124,62 @@ def slow_fetch_pnc_sbom(purl: str, product_data, build_data, sbom_data) -> None:
 
     # Save component taxonomy
     root_component.save_component_taxonomy()
+
+
+@app.task(base=Singleton, autoretry_for=RETRYABLE_ERRORS, retry_kwargs=RETRY_KWARGS)
+def slow_handle_pnc_errata_released(erratum_id: int, erratum_status: str) -> None:
+    # Check that the erratum is released
+    if erratum_status != "SHIPPED_LIVE":
+        raise ValueError(f"Invalid status {erratum_status} for erratum {erratum_id}")
+
+    # Get the purl(s) of the related component(s) from the erratum's Notes field
+    et = ErrataTool()
+    notes = et.get_erratum_notes(erratum_id)
+
+    related_purls = set()
+    for ref in notes.get("manifest", {}).get("refs", {}):
```

Comment:
The last bit looks a bit off, but I might be confused. You have `.get("refs", {})` which implies refs is a dict. Then you iterate over each ref in the refs key, but iterating over a dict only gives you the key names.

So I don't think the check below for `if ref["type"] == "purl"` is going to work. Maybe refs is a dict with string keys and nested dicts as values? In that case, below should work:
```suggestion
    for ref in notes.get("manifest", {}).get("refs", {}).values():
```
Or maybe the default for `.get()` should be an empty tuple / list instead of an empty dict, to match the real data.

Reply:
Good catch, the default for refs should have been a list, not a dict.